### DISCUSSION TITLE
New Design?

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#6782] [Ideas]
Why are custom layouts being removed, now the option for certain layouts for certain screens is gone.

**Comment by melohagan** [comment]
Hi @kasgithub8 

My understanding is that the layouts weren't being used by the vast majority of users, and the link configuration was hard to find. 
So the change was made to simplify this.

Another aspect is that layouts apply to all screens of a given role, so if you found that you had a screen or two that you wanted to be styled differently then you'd be a bit stuck. 
Applying layouts on a per screen basis gives you more flexibility, but you can still duplicate screens to copy the layout from an existing screen.

Is there something in particular that you're having issues with?

**Reply by kasgithub8** [reply]
My main Issue right now is different Layouts for different Roles. "Power" roles should be able to see more screens f.e..

**Reply by melohagan** [reply]
You can assign a role to each screen: https://docs.budibase.com/docs/screens#access-level

**Reply by kasgithub8** [reply]
Yes, I know we can. But the navigation bar was so much better with custom layouts. I get that they were trying to "simplify" it, but this is the perfect example of making it too simple.

**Reply by melohagan** [reply]
Is there something you can't do with the navigation bar that you could do before?
https://docs.budibase.com/docs/navigation

**Reply by kasgithub8** [reply]
Nah, I got it now. Thanks tho!

**Reply by kasgithub8** [reply]
@melohagan Do you have any idea why I'm not able to change the Screen Width on a form screen? Medium, Large and Max are all the same size.

**Reply by melohagan** [reply]
The width in the navigation panel refers to the width of the navigation bar. If you watch the video around the 10 second mark it shows the effect. 

To set the screen size you can set the screen padding (left and right).
![Screenshot 2022-07-20 at 10 03 43](https://user-images.githubusercontent.com/101575380/179942925-f0505780-13ff-48b3-afa7-6e6497e1e5a1.png)

**Reply by kasgithub8** [reply]
I know. I'm talking about this right here:
![image](https://user-images.githubusercontent.com/95284278/179945691-e34f2a85-22e6-462c-9a3d-b6693c9efd22.png)

**Comment by micebrain** [comment]
In my case, layouts are used to group different set of functionalities and also to add elements like icon. Yes I can copy and paste the layout design from screen to screen, but this feels like a regression because instead of reusability we introduced a less efficient and 
 error prone process - any change in layout means every single screen must be changed manually.
Is is possible to reinstate this feature?

**Reply by kasgithub8** [reply]
@micebrain I agree.

**Reply by developer-uhwi** [reply]
Custom layouts was one of the more useful features, it was a convenient way of building multiple pages with common components, now this is very hard to do with the current UI.

Currently being forced to use one layout for all pages. In previous release, I was able to have a different navigation links based on the module I was in e.g separate administration navigation vs general navigation.

How do I implement such a scenario now, seems I will have to build custom navigation, hide the default navigation on pages, and copy and paste the custom navigation to the pages, where it will be located.

So if I have a custom nav on 10 pages, and I change this custom nav, I will have to edit the nav on all 10 pages.
Not cool Budibase.

Custom Layouts are really good for defining separate modules in your applications.

What to do now????

**Comment by validlogin** [comment]
This does seem like an odd change to make. It's basically removing functionality and replacing it with.. well, less features. It is nice that we can now select a home page; however, for my use case it removes quite a few things that were very helpful. Our layout had a footer that would show the current version of the app, which would be placed on all screens. To continue doing this, I would now have to add this to each and every screen. 

Another feature (and pretty vital, for me, at least) was the ability to hide links from the navigation pane depending on the user (via conditions). Most users have the same power role, but they utilize different databases and forms, so using different roles wouldn't work in this case. I don't see a way to do this with the new update. I guess I can create a homepage with cards that have the same effect, but it's just another step to take.. and it also defeats the purpose of the navigation bar except for common/general usage links.

Just my two cents ... Budibase is still a great program, and I really look forward to new features such as the QR/Barcode feature and the PDF export feature :)

**Reply by melohagan** [reply]
Really appreciate the feedback!

**Comment by mmalessa80** [comment]
Personally I'm not fond of permanently displaying the navigation on the top or the left side when not on a phone size mobile screen. I therefore elected designing my own layouts, but it is a pain having to replicate this on every screen.

![image](https://user-images.githubusercontent.com/106758851/180515775-e8b3c54f-5bcd-4bd7-a3a7-41f80b1be9f4.png)

**Comment by weixingp** [comment]
We also have a similar use case as @validlogin where a common footer / help text is displayed at the bottom of every public screens. (And we have 20 public screens or so, will be a pain to add this into every screen. It also makes it hard to update the footer.)  

Besides that, we also have a niche use case for custom layout: displaying different layout/style when viewed under different devices. To achieve this, we would just need to duplicate the screen slot element and apply "if mobile / if desktop" to the container to display the correct container with respective style under different devices.

In order to achieve the above, we would need to duplicate everything once for the mobile screen. (At the moment, I don't think Budibase provide style configurations (like width / padding, etc.) per device)

**Comment by EuanRW** [comment]
We also have a use case for this feature similar to @weixingp and @validlogin, we need to display custom mobile/tablet layouts, and have features that need to be hidden by users depending on complex conditions.

At the moment, we are doing this manually, but it is becoming unmanageable as the number of our screens grows.

Any update on when this feature will be coming back?

**Reply by shogunpurple** [reply]
@EuanRW - this won't be coming back in it's original form. There will be something new - which is the ability to create saved screens/blocks that you can reuse and share across apps at a global level. We won't be shipping this until 2023 though

**Reply by EuanRW** [reply]
Thanks for the quick response
